### DISCUSSION TITLE
What are annotations?

### DISCUSSION THREAD

**Discussion by mark-rushakoff** [community/community#25002] [Actions]
<p>In the email reports when my Workflow fails, I see mentions of annotations, like “$xyz succeeded (0 annotations)”.</p>
<p>What are annotations? The search bar on <a href="http://help.github.com" rel="nofollow noopener">help.github.com</a> doesn’t show any relevant results.</p>

**Comment by mscoutermarsh** [comment, marked as answer]
<p>Ahh, that email is confusing.</p>
<p><a href="https://help.github.com/en/articles/about-status-checks" rel="nofollow noopener">https://help.github.com/en/articles/about-status-checks</a></p>
<p>See the comments on the lines of code? That’s an annotation. We have an API that allows annotations to be created during a CI run.</p>
<p><img src="https://user-images.githubusercontent.com/155044/181088096-36eb1d45-5b7d-4f4f-9721-185effbb9115.png" alt="checks-detailed" data-base62-sha1="wu0UPJn2XITTP3eJrKRf73oPt55" width="644" height="324"></p>
<p>They aren’t supported in Actions yet, but will be.</p>

**Comment by xt0rted** [comment]
<p><a class="mention" href="/mscoutermarsh">@mscoutermarsh</a> any idea when annotations will be supported again? I have a couple actions that need to be migrated to v2 that do this but I’m unable to create annotations with v2 actions. I can run v1 action in a v2 workflow and annotations will be created just fine, but a v2 style action doesn’t. I can create the checks run, but when I try to update it with the annotations I get a 422 response.</p>

**Comment by ankri** [comment]
<p>I have had some success (kind of) for adding annotations:</p>
<ol>
<li>Try to get the current check run by listing all the checks that are <code>in_progress</code> and then finding the current run. I wrote <a href="https://github.com/tangro/tangro-github-toolkit/blob/master/src/checks.ts#L5" rel="nofollow noopener">this helper</a>. (In this case I can make some assumptions like that the name contains “test” -  your may not. In my case I also could have returned the only check run that is in progress because I know that only one would be running)</li>
<li>Create the <a href="https://github.com/tangro/actions-test/blob/master/src/test/checkRun.ts#L76-L93" rel="nofollow noopener">annotations</a>. I am using my jest output.</li>
<li>
<a href="https://github.com/tangro/tangro-github-toolkit/blob/master/src/checks.ts#L33-L53" rel="nofollow noopener">Updating</a> the check run found in 1.</li>
</ol>
<p>With this method I can add annotations:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/ba0077e509917e8f5044a856b57eee16f8591486.png" data-download-href="/uploads/short-url/qxrOoQeZe83p1VF9l4LwIliBUSW.png?dl=1" title="Bildschirmfoto 2019-09-12 um 22.44.06.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2842920/181088098-8348147f-361b-42aa-939b-e57cf85a344b.png" alt="Bildschirmfoto 2019-09-12 um 22.44.06.png" data-base62-sha1="qxrOoQeZe83p1VF9l4LwIliBUSW" width="690" height="308" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/ba0077e509917e8f5044a856b57eee16f8591486_2_690x308.png, https://aws1.discourse-cdn.com/github/optimized/2X/b/ba0077e509917e8f5044a856b57eee16f8591486_2_1035x462.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/b/ba0077e509917e8f5044a856b57eee16f8591486_2_1380x616.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/ba0077e509917e8f5044a856b57eee16f8591486_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Bildschirmfoto 2019-09-12 um 22.44.06.png</span><span class="informations">1633×730 86.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>The real downside is, that annotation can only be viewed in files changed with the commit:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/fc706d492f8e7c6e328e17d0eb412b89eb120d54.png" data-download-href="/uploads/short-url/A1bc9HfC6jb8HDJov2h8baZu2Ne.png?dl=1" title="Bildschirmfoto 2019-09-12 um 22.48.36.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2842920/181088099-38a7c67a-1102-4045-bc6d-b081e8d9c594.png" alt="Bildschirmfoto 2019-09-12 um 22.48.36.png" data-base62-sha1="A1bc9HfC6jb8HDJov2h8baZu2Ne" width="690" height="390" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/f/fc706d492f8e7c6e328e17d0eb412b89eb120d54_2_690x390.png, https://aws1.discourse-cdn.com/github/original/2X/f/fc706d492f8e7c6e328e17d0eb412b89eb120d54.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/f/fc706d492f8e7c6e328e17d0eb412b89eb120d54.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/fc706d492f8e7c6e328e17d0eb412b89eb120d54_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Bildschirmfoto 2019-09-12 um 22.48.36.png</span><span class="informations">955×541 78.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>… So because I wanted to add the test results directly to the failing test, I am out of luck when a test fails which has not been changed with the current commit.</p>
<p>Hopefully the code samples are helpful for you.</p>

**Comment by johnsonsamuel** [comment]
<p>How can I add annotations to my npm test workflow which I have created?</p>
<p>My response after running the test:</p>
<pre><code class="lang-auto">[ { path: '.github/workflows/test.yml',  
blob\_href: 'repo...../.github/workflows/test.yml',  
start\_line: 1,  
start\_column: null,  
end\_line: 1,  
end\_column: null,  
annotation\_level: 'failure',  
title: '.github/workflows/test.yml#L1',  
message: 'Process completed with exit code 1.',  
raw\_details: null } ]  
  
</code></pre>

**Comment by johnsonsamuel** [comment]
<p>I see when my check run is running the status is from ‘queued’ to ‘completed’. I dont see ‘in_progress’. Any thoughts what I am missing ?</p>

**Comment by galak** [comment]
<p>Any update on this, or when the feature might be available?</p>

**Comment by ssbarnea** [comment]
<p>Current way annotations where implemented seems quite weird for me as I was expecting to be able to create annotation reports on disk which would be loaded by github after the build finished, avoiding the need to perform API calls during build.</p>
<p>Current implementation just creates a vendor-lockin by requesting each tool that can produce annotation like output to add specific support for GitHub.</p>
<p>I asked recently on <a href="https://stackoverflow.com/questions/62516223/how-can-i-produce-github-annotations-by-creating-report-files-on-disk" rel="nofollow noopener">https://stackoverflow.com/questions/62516223/how-can-i-produce-github-annotations-by-creating-report-files-on-disk</a> as I suppose I am not the first one that asks himself about lack of portability.</p>
<p>Do we have any file format that we can use for annotations and rely on a generic reported to send them to github?</p>

**Comment by jackfirth** [comment]
<p>Are there any updates on this? I’d like to make some annotations from an action.</p>

**Comment by Khush1694** [comment]
Any idea on how can we create links to the annotations. I want to add link to the annotations which is created using check run update API in comments but not able to figure out how can I get annotation id.
